### PR TITLE
Perform scratch builds in parallel

### PR DESCRIPTION
**Pull request by ehelms**
I am wondering if we can do this now and get better performance and more obvious output.

### CODE REVIEW

**Comment by evgeni** [comment]
Can we limit the parallelization somehow? I don't think doing 20+ packages in parallel (a usual Rails upgrade) should yield good results

File: theforeman.org/pipelines/test/rpm_packaging.groovy [hunk 1]
```diff
@@ -133,9 +133,14 @@ pipeline {
                 expression { packages_to_build }
             }
             steps {
+                for(int i = 0; i < packages_to_build.size(); i++) {
+                    def index = i
+                    packages[packages_to_build[index]] = {
```
Comment (ekohl) [inline theforeman.org/pipelines/test/rpm_packaging.groovy:138]: I'd prefer:
```suggestion
                packages = [:]
                packages_to_build.each { package ->
                    packages[package] = {
```

**Review by ekohl — COMMENTED** [review]
Koji does have queuing so perhaps that covers it enough? In theory it also means we could add a builder on days where we'd have many build jobs, though in practice that's unlikely to be something we need to optimize for.

Or are you worried about the Jenkins workers. In that case we could also look into implementing the concurrency in obal, given Ansible can also handle some concurrency.

**Comment by evgeni** [comment]
I'm mostly worried about Koji starving itself.
When you trigger a build it contains multiple steps (for scratch: rebuild srpm, build arch, for release: rebuild srpm, build arch, tag, regen all affected repos).

This PR only touches scratch, so it's not *that* dangerous (srpm and arch don't do too much locking), so we might just wait for the individual results longer. But if we apply the same logic to release, I suspect to see the overall throughput to go down as jobs wait on repo regens etc.

**Pull request closed without merging**